src/AssetsPage/AssetsTable/AssetsTable.test.jsx

```diff
@@ -24,31 +28,335 @@ const defaultProps = {
+
+    beforeEach(() => {
+      wrapper = mount(
+        <AssetsTable
+          {...defaultProps}
+        />,
+      );
+
+      trashButtons = wrapper.find('button').filterWhere(button => button.hasClass('fa-trash'));
+    });
+    it('sets modalOpen to true', () => {
+      trashButtons.at(0).simulate('click');
+      expect(wrapper.state('modalOpen')).toEqual(true);
+    });
+    it('sets assetToDelete correctly', () => {
+      const trashButtonAriaLabel = `Delete ${defaultProps.assetsList[0].display_name}`;
+      expect(trashButtons.at(0).prop('aria-label')).toEqual(trashButtonAriaLabel);
+
+      trashButtons.at(0).simulate('click');
+      expect(wrapper.state('assetToDelete')).toEqual(defaultProps.assetsList[0]);
+    });
+    it('sets elementToFocusOnModalClose correctly', () => {
+      trashButtons.at(0).simulate('click');
+
+      expect(trashButtons.at(0).matchesElement(wrapper.state('elementToFocusOnModalClose'))).toEqual(true);
+    });
+  });
+  describe('focus', () => {
+    let closeButton;
+    let deleteButtons;
+    let modal;
+    let trashButtons;
+    let mockDeleteAsset;
+
+    beforeEach(() => {
+      wrapper = mount(
+        <AssetsTable
+          {...defaultProps}
+        />,
+      );
+
+      deleteButtons = wrapper.find('button');
+      trashButtons = deleteButtons.find('button').filterWhere(button => button.hasClass('fa-trash'));
+
+      modal = wrapper.find('[role="dialog"]');
+      closeButton = modal.find('button').filterWhere(button => button.matchesElement(<button><span>&times;</span></button>));
+    });
+
+    it('moves from modal to trashcan on modal close', () => {
+      trashButtons.at(0).simulate('click');
+
+      expect(closeButton.at(0).matchesElement(document.activeElement)).toEqual(true);
+
+      closeButton.at(0).simulate('click');
+
+      expect(trashButtons.at(0).matchesElement(document.activeElement)).toEqual(true);
+    });
+
+    it('moves from modal to status alert on asset delete', () => {
+      const deleteButton = deleteButtons.filterWhere(button =>
+        button.matchesElement(<button>Yes, delete.</button>));
+
+      const statusAlert = wrapper.find('StatusAlert');
+      const closeStatusAlertButton = statusAlert.find('button').filterWhere(button => button.matchesElement(<button><span>&times;</span></button>));
+
+      trashButtons.at(0).simulate('click');
+
+      expect(closeButton.at(0).matchesElement(document.activeElement)).toEqual(true);
+
+      deleteButton.at(0).simulate('click');
+
+      expect(closeStatusAlertButton.at(0).matchesElement(document.activeElement)).toEqual(true);
+    });
+    it('moves to correct asset trashcan icon after first asset deleted', () => {
+      const assetToDeleteId = defaultProps.assetsList[0].id;
+
+      mockDeleteAsset = jest.fn(() => {
+        wrapper.setProps({ assetsList: defaultProps.assetsList.filter(asset => asset.id !== assetToDeleteId), assetsStatus: { type: 'DELETE_ASSET_SUCCESS' } });
+      });
```

Comment:
Can this `mockDeleteAsset` move to a more general location? It's duplicated below as well, or is there a reason we would want to keep it separated out?

Reply:
Ideally, I'd move it up into the beforeEach, but because assetToDeleteId differs between the tests, I need to re-initialize the function so that it includes the value of assetToDeleteId that's specific to that test. Can you think of a way to abstract that away?

Reply:
Could we curry it? Define a function (in some utility location) that takes an id, and returns this no-parameters function?

Just spitballing here, I'm okay with it as-is.